private/buf/cmd/buf/command/build/build.go

```diff
@@ -150,6 +153,12 @@ func run(
 	if err != nil {
 		return fmt.Errorf("--%s: %v", outputFlagName, err)
 	}
+	if len(flags.Types) > 0 {
```

Comment:
I don't know if this is the right place to add this or if it needs to go somewhere else (like bufcli.NewImageForSource). Added here for now so its usable from the cli.

Reply:
I think optimally this would be in `bufcli.NewImageForSource`, but there's a wider UI discussion about whether we want this flag somehow exposed in other commands, for example `buf breaking`. If we do, then we probably want to make this logic common. If we don't, then here is fine.

Reply:
I can see how this might be useful for other commands, generation seems like the most natural. Perhaps we should make the flag hidden till this is thought through a little bit more - my main need is in the GetImage api for now and this was primarily nice last minute addition so we could more easily play with it. My vote would be to keep it here (perhaps hidden) and follow up with adding it to all places in the CLI where it makes sense with proper changelog in a follow up.